Dalamud/Game/Chat/SeStringHandling/SeString.cs

```diff
@@ -100,5 +103,43 @@ public byte[] Encode()
 
             return messageBytes.ToArray();
         }
+
+        /// <summary>
+        /// Serializes the SeString to json
+        /// </summary>
+        /// <returns>An json representation of this object</returns>
+        public string ToJson()
+        {
+            return JsonConvert.SerializeObject(this, Formatting.Indented, new JsonSerializerSettings()
+            {
+                PreserveReferencesHandling = PreserveReferencesHandling.Objects,
+                ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
+                TypeNameHandling = TypeNameHandling.Auto
+            });
+        }
+
+        /// <summary>
+        /// Creates a SeString from a json. (For testing - not recommended for production use.)
+        /// </summary>
+        /// <param name="json">A serialized SeString produced by ToJson() <see cref="ToJson"/></param>
+        /// <param name="dataManager">An initialized instance of DataManager for Lumina queries.</param>
+        /// <returns>A SeString initialized with values from the json</returns>
+        public static SeString FromJson(string json, DataManager dataManager)
+        {
+            var s = JsonConvert.DeserializeObject<SeString>(json, new JsonSerializerSettings
+            {
+                PreserveReferencesHandling = PreserveReferencesHandling.Objects,
+                TypeNameHandling = TypeNameHandling.Auto,
+                ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor
+            });
+
+            foreach(var payload in s.Payloads)
+            {
+                var dataResolver = payload.GetType().GetField("DataResolver", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
```

Comment:
Please make DataResolver public instead of using reflection, other than that, looks good!

Reply:
Done!